packages/components/src/Tree/withTreeGesture.test.js

```diff
@@ -0,0 +1,219 @@
+};
+
+const ComponentWithGesture = withTreeGesture(Tree);
+
+function getSelector({ level, posinset }) {
+	return `li[aria-level=${level}][aria-posinset=${posinset}]`;
+}
+
+describe('TreeGesture HOC', () => {
+	it('should wrap Tree component', () => {
+		// then
+		expect(ComponentWithGesture.displayName).toBe('TreeGesture(Tree)');
+	});
+
+	it('should select item on enter keydown', () => {
+		// given
+		const props = {
+			...treeProps,
+			onSelect: jest.fn(),
+		};
+		const wrapper = mount(<ComponentWithGesture {...props} />);
+		const event = { keyCode: keycode.codes.enter, stopPropagation: jest.fn() };
+		expect(props.onSelect).not.toBeCalled();
+
+		// when
+		wrapper.find(getSelector({ level: 0, posinset: 0 })).simulate('keydown', event);
+
+		// then
+		expect(event.stopPropagation).toBeCalled();
+		expect(props.onSelect).toBeCalledWith(expect.anything(), props.items[0]);
+	});
+
+	it('should select item on space keydown', () => {
+		// given
+		const props = {
+			...treeProps,
+			onSelect: jest.fn(),
+		};
+		const wrapper = mount(<ComponentWithGesture {...props} />);
+		const event = { keyCode: keycode.codes.space, stopPropagation: jest.fn() };
+		expect(props.onSelect).not.toBeCalled();
+
+		// when
+		wrapper.find(getSelector({ level: 0, posinset: 0 })).simulate('keydown', event);
+
+		// then
+		expect(event.stopPropagation).toBeCalled();
+		expect(props.onSelect).toBeCalledWith(expect.anything(), props.items[0]);
+	});
+
+	it('should toggle opened item on left keydown', () => {
+		// given
+		const props = {
+			...treeProps,
+			onToggle: jest.fn(),
+		};
+		const wrapper = mount(<ComponentWithGesture {...props} />);
+		const event = { keyCode: keycode.codes.left, stopPropagation: jest.fn() };
+		expect(props.onToggle).not.toBeCalled();
+
+		// when
+		wrapper.find(getSelector({ level: 0, posinset: 1 })).simulate('keydown', event);
+
+		// then
+		expect(event.stopPropagation).toBeCalled();
+		expect(props.onToggle).toBeCalledWith(expect.anything(), props.items[1]);
+	});
+
+	it('should toggle closed item on right keydown', () => {
+		// given
+		const props = {
+			...treeProps,
+			onToggle: jest.fn(),
+		};
+		const wrapper = mount(<ComponentWithGesture {...props} />);
+		const event = { keyCode: keycode.codes.right, stopPropagation: jest.fn() };
+		expect(props.onToggle).not.toBeCalled();
+
+		// when
+		wrapper.find(getSelector({ level: 1, posinset: 1 })).simulate('keydown', event);
+
+		// then
+		expect(event.stopPropagation).toBeCalled();
+		expect(props.onToggle).toBeCalledWith(expect.anything(), props.items[1].children[1]);
+	});
+
+	it('should open all siblings on * keydown', () => {
+		// given
+		const items = treeProps.items.slice();
+		items[0] = { ...items[0], siblings: items };
+		const props = {
+			...treeProps,
+			items,
+			onToggleAllSiblings: jest.fn(),
+		};
+		const wrapper = mount(<ComponentWithGesture {...props} />);
+		const event = {
+			keyCode: keycode.codes['numpad *'],
```

Comment:
this is wrong, the code use the native event, not the numpad, so why it here ?

Reply:
Removed